test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ModelStateDictionaryExtensionsTest.cs

```diff
@@ -79,6 +79,24 @@ public void AddModelError_ForNotModelsExpression_AddsExpectedMessage()
 
         [Fact]
         public void AddModelError_ForSingleExpression_AddsExpectedException()
+        {
+            // Arrange
+            var dictionary = new ModelStateDictionary();
+            var exception = new Exception();
+
+            // Act
+            dictionary.AddModelError<TestModel>(model => model.Text, exception);
```

Comment:
Type should be inferred. Please remove the explicit type parameter unless the compiler complains.

Same elsewhere in this test class.

Reply:
Do you mean remove `<TestModel>`? The compiler tries to convert the expression to a string in that case (to try to use the overload with string key as the first param)

Reply:
Yes, I meant removing `<TestModel>`. Think I see what's going on: `TestModel` can't be inferred because it's used on the LHS of the expression. The only way to remove `<TestModel>` would be `dictionary.AddModelError((TestModel model) => model.Text, exception);` and that's just as verbose. Leave this as you've got it.